src/main/java/mekanism/client/key/MekanismKeyHandler.java

```diff
@@ -63,9 +67,33 @@ public static void registerKeybindings() {
 
     private static void handlePotentialModeItem(EquipmentSlot slot) {
         Player player = Minecraft.getInstance().player;
-        if (player != null && IModeItem.isModeItem(player, slot)) {
-            Mekanism.packetHandler().sendToServer(new PacketModeChange(slot, player.isShiftKeyDown()));
-            SoundHandler.playSound(MekanismSounds.HYDRAULIC);
+        if (player != null) {
+            if (IModeItem.isModeItem(player, slot)) {
+                Mekanism.packetHandler().sendToServer(new PacketModeChange(slot, player.isShiftKeyDown()));
+                SoundHandler.playSound(MekanismSounds.HYDRAULIC);
+            } else if (Mekanism.hooks.CuriosLoaded) {
+                CuriosIntegration.findCurio(player, s -> s.canEquip(slot, player))
+                        .stream()
+                        .filter(r -> {
+                            final boolean isMode = IModeItem.isModeItem(r.stack(), slot);
+                            if (r.stack().getItem() instanceof IGasItem gas) {
+                                return isMode && !gas.getGas(r.stack()).isEmpty();
+                            }
+                            return isMode;
+                        }).filter(r -> {
+							ItemStack stack = r.stack();
+							if (IModeItem.isModeItem(stack, slot)) {
+								return !(stack.getItem() instanceof IGasItem item) || !item.getGas(stack).isEmpty();
+							}
+							return false;
+						})
```

Comment:
Spacing is super weird here (due to it using tabs rather than spaces). Also like you have the filter taking place twice instead of replacing the existing one you had?